### ISSUE TITLE
The results of the paper cannot be reproduced

### ISSUE THREAD

**Issue by hukuda222** [DavidFanzz/llm_decoding#2]
Thank you for sharing the code.

When I look at run.sh, I understand approximately how to run the program, but when I actually run it, I cannot reproduce the results of the paper. For example, if you use llama2-7b to infer commonsense-qa with greedy search, I got the following results. The actual score was 21.05 while in the paper it was 62.96.

For each decoding mehotd and task, could you please tell me how to reproduce the results of the paper?

train command:
```
python3 generate.py \
    --decoding_method greedy\
    --infile ./data/commonsense_qa/Llama2_input.jsonl\
    --outfile ./results/commonsense_qa/Llama2_decapoda_greedy.jsonl\
    --gpus_per_model 1\
    --world_size 1\
    --batch_size 32\
    --max_new_tokens 512\
    --model_name_or_path baffo32/decapoda-research-llama-7B-hf
```

evaluation command:
```
python3 evaluation.py\
                --model Llama2 \
                --task_name commonsense_qa\
                --load_generations_path ./results/commonsense_qa/Llama2 _decapoda_greedy.jsonl\
                --metric_output_path ./results/commonsense_qa_results.jsonl\
                --allow_code_execution\
                --sft
```

result:
```
{"results": {"exact_match": 0.21048321048321048, "match_template": 1.0}, "model": "Llama2", "task_name": "commonsense_qa", "allow_code_execution": true, "load_generations_path": "./results/commonsense_qa/Llama2_decapoda_greedy.jsonl", "metric_output_path": "./results/commonsense_qa_results.jsonl", "check_references": false, "bertscore": false, "comet": false, "postprocessed_output_path": null, "sft": true}
```

**Comment by DavidFanzz** [comment]
You mentioned achieving only ~20% accuracy on the CommonsenseQA dataset. Since CommonsenseQA is a 5-choice multiple-choice task, I’d recommend first checking whether your model outputs are correctly generating the CoT reasoning steps and final answers. If they aren’t, it’s likely a model generation issue.

If the model output seems reasonable, another likely cause is evaluation mismatch. I noticed from your command:
```
python3 evaluation.py \
    --model Llama2 \
    --task_name commonsense_qa \
    --load_generations_path ./results/commonsense_qa/Llama2_decapoda_greedy.jsonl \
    --metric_output_path ./results/commonsense_qa_results.jsonl \
    --allow_code_execution \
    --sft
```
You're using the --sft tag during evaluation, which is intended for fine-tuned models (like chat-tuned ones). However, from your generation command, it appears you're using the base pretrained model (baffo32/decapoda-research-llama-7B-hf). In this case, the --sft tag should be removed, as it changes how the answers are parsed (e.g., matching parentheses and special tokens).
Please try rerunning the evaluation without --sft:

```

python3 evaluation.py \
    --model Llama2 \
    --task_name commonsense_qa \
    --load_generations_path ./results/commonsense_qa/Llama2_decapoda_greedy.jsonl \
    --metric_output_path ./results/commonsense_qa_results.jsonl \
    --allow_code_execution

```
